src/main/scala/io/iohk/ethereum/consensus/Consensus.scala

```diff
@@ -67,3 +74,44 @@ trait Consensus {
 //  private def attemptToSetNewBestBranch(branch: UpdatedBranch): Either[BlockExecutionError, Boolean] = ???
 
 }
+
+object Consensus {
+  /* This return type for consensus is probably overcomplicated for now because some information is needed
+   * to keep the compatibility with the current code (particularly for the block queue handling), and be able
+   * to translate the values to BlockImportResult.
+   * In particular:
+   *  - `blockToEnqueue` fields won't be needed if the block are already stored in memory
+   *  - The distinction between ExtendedCurrentBestBranch and SelectedNewBestBranch won't really be useful
+   *  because there will be no need to put back the old branch into the block queue in case of reorganisation
+   *  - `ConsensusErrorDueToMissingNode` and  `ConsensusError` would mean that the application is in an
+   *  inconsistent state. Unless there is a reason to think that mantis would self heal when that happens, I
+   *  don't think there is a reason to add them here.
+   */
+
+  sealed trait ConsensusResult
+
+  /** The new branch was selected and it extended the best branch. */
+  case class ExtendedCurrentBestBranch(blockImportData: List[BlockData]) extends ConsensusResult
+
+  /** The new branch was selected and it extended the best branch, but it did not execute completely. */
+  case class ExtendedCurrentBestBranchPartially(blockImportData: List[BlockData], failureBranch: BranchExecutionFailure)
```

Comment:
from the name I'd understand that the branch was partially extended, but instead it holds the branch that was extended but not executed yet. So I'd suggest `PendingExtendedCurrentBestBranch` or `NotVerifiedExtendedCurrentBestBranch`

Reply:
It holds the part of the branch that was executed in the first parameter `blockImportData` and in the second parameter `failureBranch` the part of the branch that failed the execution. The last block from `blockImportData` is saved as best block, so the current best chain get extended until that point

Reply:
then `ExtendedCurrentBestBranchWithDiscardedPart` or `ExtendedCurrentBestBranchWithFailedPart`?

Reply:
I really think `partially` serves it purpose well here. Would `ExtendedPartiallyCurrentBestBranch` be more clear?

Reply:
if the name is clear for everyone but me - just leave it as is. I know now what it means